crates/extensions/tedge_file_system_ext/src/lib.rs

```diff
@@ -121,9 +122,14 @@ impl Actor for FsWatchActor {
     }
 
     async fn run(&mut self) -> Result<(), RuntimeError> {
-        let mut fs_notify = NotifyStream::try_default().unwrap();
+        let mut fs_notify = NotifyStream::try_default().map_err(Box::new)?;
         for (watch_path, _) in self.messages.get_watch_dirs().iter() {
-            fs_notify.add_watcher(watch_path).unwrap();
+            if let Err(err) = fs_notify.add_watcher(watch_path) {
+                error!(
```

Comment:
Although this approach is less invasive I'm starting to doubt if this is the right thing to do.

The issue with it is that, we let the system continue in a broken state by silently logging the issue, which users may not notice immediately. If this actor fails to watch a directly that another actor has requested for, it is a fatal failure, in my opinion. So crashing the daemon is probably the better option I feel, so that the user immediately notices that something is wrong.

Ideally, we should have a mechanism in the actor framework where this actor can communicate this failure to the requesting actor via an error and let that actor itself decide whether to crash or not.

Reply:
I agree.